File: 1-easy/problems/1991-find-the-middle-index-in-array/solution.ts
function findMiddleIndex(nums: any): boolean | number | string | any {
  """
  :type nums: List[int]
  :rtype: int
  """
  total = sum(nums)
  accu = 0
  for (i, x in enumerate(nums)) {
      if (accu*2 == total-x) {
          return i;
      accu += x
  return -1;
}

export { findMiddleIndex };

File: 1-easy/problems/1991-find-the-middle-index-in-array/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def findMiddleIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        total = sum(nums)
        accu = 0
        for i, x in enumerate(nums):
            if accu*2 == total-x:
                return i
            accu += x
        return -1
